apps/gallery/js/ImageEditor.js

```diff
@@ -915,6 +908,22 @@ ImageEditor.prototype.resize = function() {
   }
 };
 
+// Reset and generate new preview using the updated dimensions
+// in this.source object
+ImageEditor.prototype.requestNewPreview = function(callback) {
+  if (this.preview.src) {
+    URL.revokeObjectURL(this.preview.src);
+    this.preview.removeAttribute('src');
+    $('edit-enhance-button').classList.add('disabled');
+  }
+
+  this.edit(function() {
+    if (callback) {
+      callback();
+    }
+  });
+};
```

Comment:
Okay, I see now that requestNewPreview() really is just the same as calling resetPreview and then calling edit, right?  In some places this saves you a line of code. But in others (like the case where the user cancels the edit) you end up with extra code because you could no longer just have a single edit() call after the switch statement and had to add edits to the exposure and filters case.  

I prefer having these as two separate functions because it allows us to invalidate the preview without forcing a new one to be created right away. If you called resetPreview() twice (for some reason) before calling edit() no extra work is done. But if you call requestNewPreview() twice, it redraws twice.

Not really a big deal. I slightly prefer having separate functions. If you do want to stick with this one function, though, I recommend you rename to displayNewPreview() to indicate that it does actually display the new preview.